packages/meditrak-server/src/routes/userAccounts/RegisterUserAccounts.js

```diff
@@ -87,6 +89,31 @@ export class RegisterUserAccounts extends CreateUserAccounts {
       password,
     });
 
+    if (countryCode && permissionGroupName) {
+      const country = await this.models.entity.findOne({
+        code: countryCode,
+        type: 'country',
+      });
+
+      const permissionGroup = await this.models.permissionGroup.findOne({
+        name: permissionGroupName,
+      });
+
+      if (!permissionGroup) {
+        throw new Error(`No such permission group: ${permissionGroupName}`);
+      }
+
+      if (!country) {
+        throw new Error(`No such country: ${countryCode}`);
+      }
+
+      await this.models.userEntityPermission.findOrCreate({
+        user_id: userId,
+        entity_id: country.id,
+        permission_group_id: permissionGroup.id,
+      });
```

Comment:
Need some sort of check on the permission group that the user is registering against. Currently someone could send a registration with `BES Admin` and grant themselves access to everything.

Reply:
Good point. Any ideas about the best way of doing this? I could add a `assertBESAdminAccess` before adding the extra permissions and then add `BES Admin` permission to the lesmis-server user?

Reply:
I think we generally want to avoid giving user's `BES Admin` unnecessarily.

One idea would be rather than passing through a country/permission pair we could pass through the service that the user is registering for. So `tupaia` or `lesmis` and then look up a base country/permission pair from that.